### PR TITLE
Add pointer overloads for Avx2.BroadcastScalarToVector128

### PR DESCRIPTION
**Pull request by fiigii**
close https://github.com/dotnet/corefx/issues/30266

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs [hunk 1]
```diff
@@ -252,6 +252,50 @@ public abstract class Avx2 : Avx
         /// </summary>
         public static Vector128<T> BroadcastScalarToVector128<T>(Vector128<T> value) where T : struct { throw new PlatformNotSupportedException(); }
 
+        /// <summary>
+        /// __m128i _mm_broadcastb_epi8 (__m128i a)
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs]: nit: We've left comments elsewhere when the API exposed doesn't have a corresponding native API

Comment (fiigii) [reply src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs]: Thanks, will add.

**Comment by fiigii** [comment]
@CarolEidt @eerhardt

**Comment by fiigii** [comment]
After this PR and https://github.com/dotnet/coreclr/pull/19420 gets merged, I will submit a PR to implement all the remaining AVX2 intrinsic in JIT.

**Comment by fiigii** [comment]
Added comments and disabled tests of `BroadcastScalarToVector128<T>` that have ambiguities from the new APIs.

File: src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs [hunk 2]
```diff
@@ -252,6 +252,66 @@ public abstract class Avx2 : Avx
...
+        /// The above native signature does not directly correspond to the managed signature. 
+
+        /// </summary>
+        public static unsafe Vector128<sbyte> BroadcastScalarToVector128(sbyte* source) { throw new PlatformNotSupportedException(); }
+
+        /// <summary>
+        /// __m128i _mm_broadcastw_epi16 (__m128i a)
+        ///   VPBROADCASTW xmm, m16
+        /// The above native signature does not directly correspond to the managed signature. 
+
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs]: nit: stray newlines in the middle of the doc comments for most of these.

File: src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs [hunk 3]
```diff
@@ -252,6 +252,60 @@ public abstract class Avx2 : Avx
...
+        ///   VPBROADCASTB xmm, m8
+        /// The above native signature does not directly correspond to the managed signature. 
+        /// We provide this additional overload for the lack of pointers to managed.
+        /// </summary>
+        public static unsafe Vector128<byte> BroadcastScalarToVector128(byte* source) { throw new PlatformNotSupportedException(); }
+        /// <summary>
+        /// __m128i _mm_broadcastb_epi8 (__m128i a)
+        ///   VPBROADCASTB xmm, m8
+        /// The above native signature does not directly correspond to the managed signature. 
+
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs]: nit: Still one remaining

**Comment by tannergooding** [comment]
Thanks for the work @fiigii. Feel free to tag me when the tests complete and I'll get this merged.

**Comment by fiigii** [comment]
@dotnet-bot test this please

**Comment by fiigii** [comment]
CI is still stuck... Can we merge this and https://github.com/dotnet/coreclr/pull/19420? I have submitted corefx PR.

**Comment by CarolEidt** [comment]
Many PRs are stuck on testing, but I'm not comfortable with the idea that we should just merge without completing at least the xarch testing that's in the main CI set.

**Comment by tannergooding** [comment]
Merging, Two of the pending jobs actually completed. The other was an out of disk space issue, but a similar job (testing additional functionality) succeeded.

For the additional failures, they are also unrelated CI failures and had been previously passing (they were rerun due to a new commit which cleaned up some code comments).

**Comment by tannergooding** [comment]
@CarolEidt, saw your comment after I hit merge. The xarch jobs were all passing, CI just didn't update.

All the jobs had also been passing before the code comment cleanup I asked @fiigii to do, so the minor cleanup that did occur shouldn't have caused any additional regressions in the remaining jobs (which were mostly the perf ones).

**Comment by CarolEidt** [comment]
@tannergooding - thanks for the clarification, and for the work of tracking down job status.

**Pull request merged**